docs/release process.md

```diff
@@ -0,0 +1,45 @@
+
+This document describes the Neurobooth release process.
+
+The goals for this process are:
+1. To ensure consistency in Neurobooth deployments such that every environment is associated with a known release version. 
+2. To ensure the code tested in staging is identical to the code deployed to production.
```

Comment:
Nitpick: This is mostly true, though we want to be able to easily (and temporarily) move ahead of production in staging when testing new code. It can be invaluable to quickly test something out hardware-wise to see if it is feasible.

Reply:
Looks like the pre-releases you mentioned below may solve this for us assuming deployment is mostly automated / not a headache.

Reply:
> Looks like the pre-releases you mentioned below may solve this for us assuming deployment is mostly automated / not a headache.

It would be great if we could walk through a deployment some Friday.  IIUC, you currently deploy by pulling from master. How different is that from doing a git checkout specifying the tag associated with the release in addition to the branch?

Reply:
> Staging is usually ahead of production. The point here is not that they're always in the same state, but rather that we haven't accidentally deployed code into production that is different from what was (hopefully, rigorously) tested in staging.

Agreed&mdash;I think I misread point 2 without the word "tested".

> It would be great if we could walk through a deployment some Friday. IIUC, you currently deploy by pulling from master. How different is that from doing a git checkout specifying the tag associated with the release in addition to the branch?

Sure we can walk through it! It's really just logging into each machine, making sure master is checkout out, and then doing a pull to sync the code. I think a checkout as specified would end up being very similar.